SignalServiceKit/src/Util/AppContext.h

```diff
@@ -10,8 +10,16 @@ typedef void (^BackgroundTaskExpirationHandler)(void);
 
 @protocol AppContext <NSObject>
 
-- (BOOL)isMainApp;
-- (BOOL)isMainAppAndActive;
+@property (nonatomic, readonly) BOOL isMainApp;
+@property (nonatomic, readonly) BOOL isMainAppAndActive;
+
+// Whether the user is using a right-to-left language like Arabic
+@property (nonatomic, readonly) BOOL isRTL;
+
+@property (nonatomic, readonly) BOOL isRunningTests;
+
+// Useful for translating coordinates to the "entire screen"
+@property (nonatomic, readonly) UIView *keyWindow;
```

Comment:
nullable, right?

Reply:
Oh huh, this doesn't actually return a window.  Let's rename it to ... `rootReferenceView`?  

I wonder if this is a good idea.  Ideally, we'll only be using this to convert from the device coordinate system to a specific view's coordinate system.  

Currently, the SAE context is returning `self.rootViewController.view` and the main app context is returning the `keyWindow`.  I'm not sure this will yield the correct behavior in all cases.  I'm worried about:

* When app supports multiple orientations (i.e. landscape), this coordinate system conversion has to take into account rotation, etc.  
* I think there may be other cases where the device coordinate system gets "complicated", e.g. ios multitasking on iPad.

I'm not sure `self.rootViewController.view` will do the right thing in these cases.

I guess for the purposes of this release let's just make sure that this works properly in landscape.

Reply:
I've renamed per your suggestion.

Zoom functionality seems to work fine in landscape, portrait and when switching between the two.